### PR TITLE
refactor(app): run setup and window creation when event loop is ready

### PR DESCRIPTION
**Pull request by lucasfernog**
### What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [x] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [ ] No

### Checklist
- [ ] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [x] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [ ] I have added a convincing reason for adding this feature, if necessary

### CODE REVIEW

**Comment by amrbashir** [comment]
Doesn't this make the setup hook obsolete? because I could just do the same logic in `RunEvent::Ready`

File: .changes/refactor-setup.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"tauri": major
+---
+
+The window creation and setup hook are now called when the event loop is ready.
```
Comment (amrbashir) [inline .changes/refactor-setup.md]:
```suggestion
[**Breaking Change**] The window creation and setup hook are now called when the event loop is ready.
```

**Comment by lucasfernog** [comment]
> Doesn't this make the setup hook obsolete? because I could just do the same logic in `RunEvent::Ready`

It does, and we might remove all of these hooks in favor of the run closure, but for now I'll avoid basic breaking changes since we're actively developing mobile.

**Comment by amrbashir** [comment]
I see, but this is still a breaking change and we should note that in the change file.

**Comment by lucasfernog** [comment]
> I see, but this is still a breaking change and we should note that in the change file.

Yeah. This is the beginning of the `next` branch too :)

**Pull request merged**